src/errors.rs

```diff
@@ -16,3 +18,36 @@ impl From<String> for SdError {
         Self(arg)
     }
 }
+
```

Comment:
I doubt a consumer will really have any need to explicitly match on any of those.
In the spirit of keeping the API surface to a sane size and not exposing too many internal details, I'd string-ify those error directly, before storing them.

Reply:
:shrug: Alright